src/oci/mod.rs

```diff
@@ -41,31 +44,31 @@ pub fn ls_layer(repo: &Repository, name: &str) -> Result<()> {
     Ok(())
 }
 
-struct ImageOp<'repo> {
-    repo: &'repo Repository,
+struct ImageOp<'repo, ObjectID: FsVerityHashValue> {
+    repo: &'repo Repository<ObjectID>,
     proxy: ImageProxy,
     img: OpenedImage,
     progress: MultiProgress,
 }
 
-fn sha256_from_descriptor(descriptor: &Descriptor) -> Result<Sha256HashValue> {
+fn sha256_from_descriptor(descriptor: &Descriptor) -> Result<Sha256Digest> {
     let Some(digest) = descriptor.as_digest_sha256() else {
         bail!("Descriptor in oci config is not sha256");
     };
     Ok(parse_sha256(digest)?)
 }
 
-fn sha256_from_digest(digest: &str) -> Result<Sha256HashValue> {
+fn sha256_from_digest(digest: &str) -> Result<Sha256Digest> {
     match digest.strip_prefix("sha256:") {
         Some(rest) => Ok(parse_sha256(rest)?),
         None => bail!("Manifest has non-sha256 digest"),
     }
 }
 
-type ContentAndVerity = (Sha256HashValue, Sha256HashValue);
+type ContentAndVerity<ObjectID> = (Sha256Digest, ObjectID);
```

Comment:
This one isn't as big of a deal being a type alias since it's not public, but worth consideration still.